src/test/java/org/reaktivity/nukleus/tcp/internal/streams/rfc793/ClientIOExceptionFromReadIT.java

```diff
@@ -118,4 +119,30 @@ public void shouldNotResetWhenProcessingEndAfterIOExceptionFromRead() throws Exc
         }
     }
 
+    @Test
+    @Specification({
+        "${route}/client/controller",
+        "${client}/client.sent.reset.and.end/client"
+    })
+    public void shouldAbortWhenProcessingEndAfterIOExceptionFromRead() throws Exception
```

Comment:
The purpose of this test class is to make sure TCP nukleus correctly reports IOException from read as an Abort frame towards the client. So it does not make sense to exercise this test script from this class. It should go in `ClientResetAndAbortIT`, which already contains a test for script "client.sent.reset" which is of course very similar. The only difference with your script is the client sends End on the stream in the other direction. To add a test for this script you should add a test method to ClientResetAndAbortIT very similar to the method `shouldShutdownOutputAndInputWhenClientSendsResetAndAbort` which you already added. In fact it can be an exact copy of that method, just renaming the method to ...End instead of ...Abort and of course changing it to point at script client.sent.reset.and.end.

Reply:
Moved the test to ClientResetAndAbortIT.